src/sentry/auth/providers/fly/provider.py

```diff
@@ -1,15 +1,122 @@
+from typing import Any, Optional
+
+from sentry import options
 from sentry.auth.partnership_config import SPONSOR_OAUTH_NAME, ChannelName
-from sentry.auth.providers.oauth2 import OAuth2Provider
+from sentry.auth.provider import MigratingIdentityId
+from sentry.auth.providers.oauth2 import OAuth2Callback, OAuth2Provider
 
+from .constants import ACCESS_TOKEN_URL, AUTHORIZE_URL
+from .views import FetchUser, FlyConfigureView, FlyOAuth2Login
 
-class FlyOAuth2Provider(OAuth2Provider):
 
+class FlyOAuth2Provider(OAuth2Provider):
     name = SPONSOR_OAUTH_NAME[ChannelName.FLY_IO]
     is_partner = True
+    access_token_url = ACCESS_TOKEN_URL
+    authorize_url = AUTHORIZE_URL
+
+    def __init__(self, org=None, **config):
+        self.org = org
+        super().__init__(**config)
+
+    def get_client_id(self):
+        return options.get("auth-fly.client-id")
+
+    def get_client_secret(self):
+        return options.get("auth-fly.client-secret")
+
+    def get_configure_view(self):
+        # Utilized from organization_auth_settings.py when configuring the app
+        # Injected into the configuration form
+        return FlyConfigureView.as_view()
 
-    # TODO: (nhsiehgit) Replace boilerplate code
-    def get_refresh_token_url(self) -> str:
-        return ""
+    def get_auth_pipeline(self):
+        return [
+            FlyOAuth2Login(client_id=self.get_client_id()),
+            OAuth2Callback(
+                access_token_url=ACCESS_TOKEN_URL,
+                client_id=self.get_client_id(),
+                client_secret=self.get_client_secret(),
+            ),
+            FetchUser(org=self.org),
+            # ConfirmEmail(),  # reference github provider
+        ]
+
+    def get_refresh_token_url(self):
+        return ACCESS_TOKEN_URL
+
+    def build_config(self, state: Any, organization: Optional[Any] = None):
```

Comment:
note: since Fly will not be configurable outside of the provider request - this step helper unnecessary/unused

Reply:
It's not abstract method, can you skip its definition completely then?

Reply:
Was debating this as well.

My thought is that at provisioning time, we might want to utilize this config constructor.
Only thing is it looks like we're serializing a different response during provisioning time.
I'll make this a classmethod for now so we don't need the provider instance at least and we can curate its usage once we're ready to test end to end